wordpress-mobile/WordPress-iOS
# [Gutenberg] Contact Support from inside editor

This feature adds the ability to:
1. Create a support ticket from the editor's Help & Support view
2. Access all support options from the editor's Help & Support view

Internal Ref: pbArwn-2IA-p2#comment-3977

### Related PRs

- https://github.com/wordpress-mobile/gutenberg-mobile/pull/3975
- https://github.com/WordPress/gutenberg/pull/34890

## To test:

### Test 1

1. Inside the editor, tap the 3-dot menu button.
2. Tap on the "Help & Support" option.
3. In the full-screen editor help center view, tap the "Contact Support" button.
4. Observe the "Contact Support" chat window.
5. Start a support session from this view and confirm it generates a Zendesk ticket.
6. Confirm that `origin:editor-help` tag is passed to Zendesk in the ticket.
7. Confirm that the event `wpios_support_new_request_viewed` appears in the internal tracking tool.

<img width="250" alt="Zendesk support view" src="https://user-images.githubusercontent.com/2092798/134103169-588df2b9-5491-47dd-901d-e20e3507503c.png" />

### Test 2

1. Inside the editor, tap the 3-dot menu button.
2. Tap on the "Help & Support" option.
3. In the full-screen editor help center view, tap the "All Support Options" button.
4. Observe the app's support view.
5. Ensure each option in the view functions as designed.

<img width="250" alt="Support view for the app" src="https://user-images.githubusercontent.com/2092798/134103284-089dfe14-db6d-455c-9da3-b899f773e27a.png" />

## Regression Notes
1. Potential unintended areas of impact
The "new Zendesk ticket" flow could be impacted.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tried out the "My Site > Avatar at the top-right > Help & Support > Contact Support" flow both from inside and outside the editor and confirmed it works OK.

3. What automated tests I added (or what prevented me from doing so)
No automated tests added. This is not a critical flow so not considered adding a UI test.

PR submission checklist:
